### ISSUE TITLE
Is it possible to request RGBA texture formats

### ISSUE THREAD

**Issue by timeinpixels** [AcademySoftwareFoundation/OpenColorIO#1685]
Hi,

Is it possible to request RGBA pixel format textures buffers for the GPU shader transform?
In Metal RGB is not supported so it would need to be manually converted to RGBA...

Thanks in advance,
Tom

**Comment by remia** [comment]
Sorry for the late reply, just noting that you can look at https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/main/src/libutils/oglapphelpers/msl.mm#L19 for an example of how this is handled in the current metal application in OCIO. Also checking with @Morteeza if there is any plan to support RGB texture at some point in Metal?

**Comment by timeinpixels** [comment]
Yes, that's my point. We need to iterate over the whole array and copy it to the new buffer. 
We could fill the RGBA buffer directly with the OCIO API call. Faster and less memory used.
